x/gamm/pool-models/balancer/amm_test.go

```diff
@@ -833,6 +833,96 @@ func TestCalcJoinPoolShares(t *testing.T) {
 			),
 			expectShares: sdk.NewInt(1250000000000 + 609374990000),
 		},
+		{
+			name:    "minimum input with two assets and minimum liquidity",
+			swapFee: sdk.MustNewDecFromStr("0"),
+			poolAssets: []balancer.PoolAsset{
+				{
+					Token:  sdk.NewInt64Coin("uosmo", 1),
+					Weight: sdk.NewInt(100),
+				},
+				{
+					Token:  sdk.NewInt64Coin("uatom", 1),
+					Weight: sdk.NewInt(100),
+				},
+			},
+			tokensIn: sdk.NewCoins(
+				sdk.NewInt64Coin("uosmo", 1),
+				sdk.NewInt64Coin("uatom", 1),
+			),
+			expectShares: sdk.NewInt(1e18).Mul(sdk.NewInt(100)),
+			expectLiq: sdk.NewCoins(
+				sdk.NewInt64Coin("uosmo", 1),
+				sdk.NewInt64Coin("uatom", 1),
+			),
+		},
+		{
+			name:    "minimum input single asset equal liquidity",
+			swapFee: sdk.MustNewDecFromStr("0"),
+			poolAssets: []balancer.PoolAsset{
+				{
+					Token:  sdk.NewInt64Coin("uosmo", 1_000_000_000_000),
+					Weight: sdk.NewInt(100),
+				},
+				{
+					Token:  sdk.NewInt64Coin("uatom", 1_000_000_000_000),
+					Weight: sdk.NewInt(100),
+				},
+			},
+			tokensIn: sdk.NewCoins(
+				sdk.NewInt64Coin("uosmo", 1),
+			),
+			expectShares: sdk.NewInt(50_000_000),
+			expectLiq: sdk.NewCoins(
+				sdk.NewInt64Coin("uosmo", 1),
+			),
+		},
+		{
+			name:    "minimum input single asset imbalanced liquidity",
```

Comment:
Let's move single asset to this variable:
https://github.com/osmosis-labs/osmosis/blob/bf8ef5b13b7f3a5ce58ad1c30cb5ded86d7927cf/x/gamm/pool-models/balancer/amm_test.go#L62

So that the test case is shared between multiple test cases that test similar functionality of different functions and methods

Reply:
yup!

Reply:
Good call!